Check new neighbors deterministically

filipbartek opened the pull request:
One-exchange neighborhood iterator rigorously checks some of the generated configurations. The check is performed with the probability 5 %, as decided by `np.random`.

Make the check deterministic (using an explicitly seeded `np.random.RandomState`) instead of relying on the default numpy random state, which need not be seeded.

Resolves #194.

mfeurer reviewed (changes requested):
Thanks a lot for this fix. Unfortunately, this currently breaks one test which has a fixture that's based on the random seed used (which now changed). Could you please have a look into that?

filipbartek commented:
> this currently breaks one test which has a fixture that's based on the random seed used (which now changed). Could you please have a look into that?
I have updated the fixture to match the current behavior of the library.
